google/cloud/firestore_v1/stream_generator.py

```diff
@@ -30,7 +30,7 @@
 T = TypeVar("T")
 
 
-class StreamGenerator(Generator[T, Any, None]):
+class StreamGenerator(Generator[T, Any, Optional[ExplainMetrics]]):
```

Comment:
> I'm aware of the overload incompatibility here, but if we change the method signature now, we risk being backward incompatible with the version we just released 😅

It looks like this passes the check:

```
def throw(self, exp=None, value=None, traceback=None) -> T:
    return self._generator.throw(exp, value, traceback)
```

This should be backwards compatible, since we're just adding new optional arguments. All existing code should be unaffected

---

Honestly though, I think removing the `=None` from `exp=None` would be best. exp is meant to be an exception, so passing None just raises a TypeError. Usually we should avoid potential breaking changes, but I don't think that applies in this case, where the default itself broken. Especially since this was such a recent addition. Let me know what you think

Reply:
You are right, we shouldn't let `exp` be optional. Just to track our discussion offline, I tried to add type annotations for all the parameters here, but it seems mypy has trouble matching it with the annotations that are present in the parent class `Generator`. So we will skip the annotation here and let mypy infer it from the parent class.

Reply:
Additionally, python 3.12 has [deprecated](https://docs.python.org/3/reference/expressions.html#generator.throw) parameters `value` and `traceback` in `generator.throw()`, so to be compatible with both 3.12 and older versions, we will just use `throw(*args, **kwargs)`.